index.html

```diff
@@ -618,6 +618,93 @@ <h4>
         </section>
       </section>
     </section>
+    <section id="automation" class="informative">
+      <h2>
+        Automation
+      </h2>
+      <p>
+        The Device Posture API pose a challenge to test authors, as fully
+        exercising interface requires physical hardware devices. To address this
+        challenge this document define [[WEBDRIVER2]]
+        <a href="https://w3c.github.io/webdriver/#dfn-extension-commands">
+        extension command</a> that allow controlling device posture like in real
+        device.
+      </p>
+      <h3>
+        Extension Commands
+      </h3>
+      <h4>
+        Set device posture
+      </h4>
+      <table class="simple">
+        <tr>
+          <th>
+            HTTP Method
+          </th>
+          <th>
+            <a href="https://w3c.github.io/webdriver/#dfn-extension-command-uri-template">
+            URI Template</a>
+          </th>
+        </tr>
+        <tr>
+          <td>
+            POST
+          </td>
+          <td>
+            window/deviceposture
+          </td>
+        </tr>
+      </table>
+      <p>
+        This <a href="https://w3c.github.io/webdriver/#dfn-extension-commands">
+        extension command</a> changes device posture to certain
+        {{DevicePostureType}}.
+      </p>
+      <table class="simple">
+        <caption>
+          Properties of the parameters argument used by this algorithm
+        </caption>
+        <tr>
+          <th>
+            Parameter name
+          </th>
+          <th>
+            Value type
+          </th>
+          <th>
+          Required
+          </th>
+        </tr>
+        <tr>
+          <td>
+            posture
+          </td>
+          <td>
+            String
+          </td>
+          <td>
+            yes
+          </td>
+        </tr>
+      </table>
+      <p>
+        The <a href="https://w3c.github.io/webdriver/#dfn-remote-end-steps">
+        remote end steps</a> are:
+      </p>
+      <ol class="algorithm">
+        <li>If |posture| is not {{DevicePostureType}}, return <a
```

Comment:
The algorithm needs to be as specific as possible. What does "be {{DevicePostureType}}" mean? Are you talking about types? Values?

Reply:
One way of writing could be 
"If |posture| is an inappropriate <a>posture</a>,"
then "<a>posture</a>" would point to https://www.w3.org/TR/device-posture/#dfn-posture
But somehow I like more when pseudo code refers to actual code (DevicePostureType enum).
Do not know if this is even bad engineering english
"If |posture| is not valid enumeration of {{DevicePostureType}},"
On the other hand I would not like that |posture| would be enum, but string as in |type| in generic sensor. I do not know if that is even possible in this interface.

Reply:
Think of it like this: at this point all you have is a JS string object called "posture". The entire DevicePostureType enum is a WebIDL construct that doesn't exist in JavaScript, as it's just used when converting to and from JS and C++ to validate strings.

One thing you could try to do is use [WebIDL's own terminology](https://webidl.spec.whatwg.org/#js-enumeration) when parsing JS enum values and say "one of DevicePostureType's enumeration values", but I'm not sure how usual that is.

The other option is to validate the value by hand: check each value, and throw an error if it does not match anything you expect.